### PR TITLE
[new release] fileutils (0.6.1)

### PR DESCRIPTION
**Pull request by gildor478**
API to manipulate files (POSIX like) and filenames

- Project page: <a href="https://github.com/gildor478/ocaml-fileutils">https://github.com/gildor478/ocaml-fileutils</a>
- Documentation: <a href="https://gildor478.github.io/ocaml-fileutils/">https://gildor478.github.io/ocaml-fileutils/</a>

### Fixed

- Migrate the build system to dune (thanks to zapashcanon).
- Migrate CHANGELOG to CHANGES for dune-release.

### CODE REVIEW

File: packages/fileutils/fileutils.0.6.1/opam [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
+  ["dune" "build" "@doc" "-p" name] {with-doc}
+]
+depends: [
+  "ocaml"
+  "base-unix"
+  "base-bytes"
+  "stdlib-shims"
+  "dune" {build & >= "1.11.0"}
```
Comment (zapashcanon) [inline packages/fileutils/fileutils.0.6.1/opam]: @gildor478 dune shouldn't be marked as `build`, sorry I didn't know it when I switched the project to dune. See #14266.

Comment (gildor478) [reply packages/fileutils/fileutils.0.6.1/opam]: I have fixed both this PR and the upstream opam files.

File: packages/fileutils/fileutils.0.6.1/opam [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+bug-reports: "https://github.com/gildor478/ocaml-fileutils/issues"
+doc: "https://gildor478.github.io/ocaml-fileutils/"
+
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
+  ["dune" "build" "@doc" "-p" name] {with-doc}
+]
+depends: [
+  "ocaml"
```
Comment (kit-ty-kate) [inline packages/fileutils/fileutils.0.6.1/opam]:
```suggestion
  "ocaml" {>= "4.03"}
```
This package fails with OCaml 4.02:
```
#=== ERROR while compiling fileutils.0.6.1 ====================================#
# context              2.0.5 | linux/x86_64 | ocaml-base-compiler.4.02.3 | pinned(https://github.com/gildor478/ocaml-fileutils/releases/download/v0.6.1/fileutils-v0.6.1.tbz)
# path                 ~/.opam/4.02/.opam-switch/build/fileutils.0.6.1
# command              ~/.opam/4.02/bin/dune build -p fileutils -j 2
# exit-code            1
# env-file             ~/.opam/log/fileutils-15-88b638.env
# output-file          ~/.opam/log/fileutils-15-88b638.out
### output ###
#     ocamldep src/.fileutils.objs/FileUtilCP.ml.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamldep.opt -modules -impl src/FileUtilCP.ml) > _build/default/src/.fileutils.objs/FileUtilCP.ml.d
# File "src/FileUtilCP.ml", line 59, characters 0-7765:
# Error: Syntax error: attributes not expected.
```

File: packages/fileutils/fileutils.0.6.1/opam [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+authors: [ "Sylvain Le Gall" ]
+homepage: "https://github.com/gildor478/ocaml-fileutils"
+license: "LGPL-2.1 with OCaml linking exception"
+dev-repo: "git+https://github.com/gildor478/ocaml-fileutils.git"
+bug-reports: "https://github.com/gildor478/ocaml-fileutils/issues"
+doc: "https://gildor478.github.io/ocaml-fileutils/"
+
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
```
Comment (kit-ty-kate) [inline packages/fileutils/fileutils.0.6.1/opam]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```

File: packages/fileutils/fileutils.0.6.1/opam [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  ["dune" "build" "@doc" "-p" name] {with-doc}
```
Comment (kit-ty-kate) [inline packages/fileutils/fileutils.0.6.1/opam]:
```suggestion
  ["dune" "build" "@doc" "-p" name "-j" jobs] {with-doc}
```

**Comment by kit-ty-kate** [comment]
Why did you remove the label? This comment hasn't been taken care of: https://github.com/ocaml/opam-repository/pull/14818#discussion_r324428051

**Comment by gildor478** [comment]
Sorry, I didn't see the 2nd comment. This should be fixed now.

**Comment by kit-ty-kate** [comment]
thanks a lot!

**Pull request merged**